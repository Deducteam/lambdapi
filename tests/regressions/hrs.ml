(** HRS file generation *)
open Common

open Handle

let () =
  Library.set_lib_root (Some "/tmp");
  Timed.(Console.verbose := 0);
  let sign = Compile.Pure.compile_file "../OK/bool.lp" in
  Export.Hrs.to_HRS Format.std_formatter sign
