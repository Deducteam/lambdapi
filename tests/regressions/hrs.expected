(FUN
A : t -> t -> t
B : t -> (t -> t) -> t
L : t -> t -> (t -> t) -> t
P : t -> (t -> t) -> t
tests_OK_bool_B : t
tests_OK_bool_bool_and : t
tests_OK_bool_bool_impl : t
tests_OK_bool_bool_neg : t
tests_OK_bool_bool_or : t
tests_OK_bool_bool_xor : t
tests_OK_bool_false : t
tests_OK_bool_true : t
)
(VAR
$x : t
$y : t -> t
$z : t
$10_0 : t
$1_0 : t
$2_0 : t
$3_0 : t
$4_0 : t
$7_0 : t
$8_0 : t
$9_0 : t
80 : t
81 : t
82 : t
83 : t
84 : t
85 : t
)
(RULES
A(L($x,$y),$z) -> $y($z),
let($x,$z,$y) -> $y($z),
A(A(tests_OK_bool_bool_and,$1_0),tests_OK_bool_false) -> tests_OK_bool_false,
A(A(tests_OK_bool_bool_and,$2_0),tests_OK_bool_true) -> $2_0,
A(A(tests_OK_bool_bool_and,tests_OK_bool_false),$3_0) -> tests_OK_bool_false,
A(A(tests_OK_bool_bool_and,tests_OK_bool_true),$4_0) -> $4_0,
tests_OK_bool_bool_impl -> L(tests_OK_bool_B,\80.L(tests_OK_bool_B,\81.A(A(tests_OK_bool_bool_or,81),A(tests_OK_bool_bool_neg,80)))),
A(tests_OK_bool_bool_neg,tests_OK_bool_false) -> tests_OK_bool_true,
A(tests_OK_bool_bool_neg,tests_OK_bool_true) -> tests_OK_bool_false,
A(A(tests_OK_bool_bool_or,$7_0),tests_OK_bool_false) -> $7_0,
A(A(tests_OK_bool_bool_or,$8_0),tests_OK_bool_true) -> tests_OK_bool_true,
A(A(tests_OK_bool_bool_or,tests_OK_bool_false),$9_0) -> $9_0,
A(A(tests_OK_bool_bool_or,tests_OK_bool_true),$10_0) -> tests_OK_bool_true,
tests_OK_bool_bool_xor -> L(tests_OK_bool_B,\82.L(tests_OK_bool_B,\83.let(tests_OK_bool_B,A(A(tests_OK_bool_bool_and,82),A(tests_OK_bool_bool_neg,83)),\84.let(tests_OK_bool_B,A(A(tests_OK_bool_bool_and,83),A(tests_OK_bool_bool_neg,82)),\85.A(A(tests_OK_bool_bool_or,84),85)))))
)
